src/main/scala/com/microsoft/partnercatalyst/fortis/spark/Pipeline.scala

```diff
@@ -0,0 +1,100 @@
+package com.microsoft.partnercatalyst.fortis.spark
+
+// TODO: remove this once transform context settings can be read from Cassandra
+import com.microsoft.partnercatalyst.fortis.spark.ProjectFortis.Settings
+
+import scala.reflect.runtime.universe.TypeTag
+import com.microsoft.partnercatalyst.fortis.spark.analyzer.{Analyzer, ExtendedFortisEvent}
+import com.microsoft.partnercatalyst.fortis.spark.dba.ConfigurationManager
+import com.microsoft.partnercatalyst.fortis.spark.dto.{Analysis, FortisEvent}
+import com.microsoft.partnercatalyst.fortis.spark.streamprovider.StreamProvider
+import com.microsoft.partnercatalyst.fortis.spark.transforms.ZipModelsProvider
+import com.microsoft.partnercatalyst.fortis.spark.transforms.image.{ImageAnalysisAuth, ImageAnalyzer}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.language.{LanguageDetector, LanguageDetectorAuth}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.{Geofence, LocationsExtractor, LocationsExtractorFactory, PlaceRecognizer}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.client.FeatureServiceClient
+import com.microsoft.partnercatalyst.fortis.spark.transforms.people.PeopleRecognizer
+import com.microsoft.partnercatalyst.fortis.spark.transforms.sentiment.{SentimentDetector, SentimentDetectorAuth}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.topic.KeywordExtractor
+import org.apache.spark.streaming.StreamingContext
+import org.apache.spark.streaming.dstream.DStream
+
+object Pipeline {
+  def apply[T: TypeTag](name: String, analyzer: Analyzer[T], ssc: StreamingContext, streamProvider: StreamProvider, configurationManager: ConfigurationManager): Option[DStream[FortisEvent]] = {
+    val configs = configurationManager.fetchStreamConfiguration(name)
+    val sourceStream = streamProvider.buildStream[T](ssc, configs)
+
+    sourceStream.map(_.transform(rdd => {
```

Comment:
That'd be fine.

In short, for each batch, we'll check Cassandra for configuration changes (every 5 to 10 seconds or so, just once on the driver). Let's say the keyword whitelist is updated. We will fetch that data from Cassandra (still within the `transform` function, and hence still on the driver), and then broadcast it (also from within `transform`). The updated data will be available for use by the `rdd` pipeline stages below through the broadcast handle from the enclosing `transform` scope. 

Note that we need to bring the data from Cassandra to the driver before broadcasting since it's only possible to broadcast data out from the driver to workers, and not from worker to workers. This isn't a drawback, but might be useful for your understanding.

In terms of efficiency of updates to already broadcasted information, unfortunately there's no concept of transferring just the delta. In practice, this should be fine since we will only rebroadcast when the user has updated terms. Also note that broadcasting itself doesn't incur transfer penalty. Each node has a cache of broadcasted data that, on miss, will fetch from the driver.